core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -271,13 +271,22 @@ object ConsumerPerformance extends LazyLogging {
 
     val brokerHostsAndPorts = options.valueOf(if (options.has(bootstrapServerOpt)) bootstrapServerOpt else brokerListOpt)
     props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerHostsAndPorts)
-    props.put(ConsumerConfig.GROUP_ID_CONFIG, options.valueOf(groupIdOpt))
-    props.put(ConsumerConfig.RECEIVE_BUFFER_CONFIG, options.valueOf(socketBufferSizeOpt).toString)
-    props.put(ConsumerConfig.MAX_PARTITION_FETCH_BYTES_CONFIG, options.valueOf(fetchSizeOpt).toString)
-    props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, if (options.has(resetBeginningOffsetOpt)) "latest" else "earliest")
-    props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, classOf[ByteArrayDeserializer])
-    props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, classOf[ByteArrayDeserializer])
-    props.put(ConsumerConfig.CHECK_CRCS_CONFIG, "false")
+    if (props.getProperty(ConsumerConfig.GROUP_ID_CONFIG) == null)
```

Comment:
Is ```props.putIfAbsent``` more better ?

Reply:
So sorry, no one replied to me before. I haven't paid attention to it for a long time

Reply:
Hi @chia7712   This is my first submission, use `props.putIfAbsent` is better, but i dont what should do next？

Reply:
> i dont what should do next？

Could you update the code (according to the reviews) and then push the changes to be a new commit. Also, please rebase your code (by trunk).

Reply:
Do you mean i need to fork apache/kafka current trunk again? if that i will delete this fork.

Reply:
I had forked once again and then updated the code (according to your suggestion). Can you take a look at this[https://github.com/apache/kafka/pull/9790](url)?